### PR TITLE
test load airports when location has changed, implemented the equals method on the Airpot class

### PR DESCRIPTION
**Pull request by rogernkosi**
… method in the Airport object

### CODE REVIEW

File: app/src/test/java/com/example/flightstimetable/ui/nearby/NearbyPresenterTest.kt [hunk 1]
```diff
@@ -144,4 +152,40 @@ class NearbyPresenterTest {
...
+        Mockito.`when`(airportApi.getAirports(apiKey,lat,long,distance)).thenReturn(Observable.just(airport))
+        given(mockLocation.latitude).willReturn(latitude)
+        given(mockLocation.longitude).willReturn(longitude)
+        mockLocation.latitude = latitude
+        mockLocation.longitude = longitude
+        presenter.locationChanged(mockLocation)
+
+        verify(view).showLoading()
+        verify(view).getAirports(airport)
+
```
Comment (N2Android) [inline app/src/test/java/com/example/flightstimetable/ui/nearby/NearbyPresenterTest.kt]: White space.

File: app/src/test/java/com/example/flightstimetable/ui/nearby/NearbyPresenterTest.kt [hunk 2]
```diff
@@ -144,4 +152,40 @@ class NearbyPresenterTest {
...
+    }
+
+    @Test
+    fun show_error_when_an_error_occurs(){
```
Comment (N2Android) [inline app/src/test/java/com/example/flightstimetable/ui/nearby/NearbyPresenterTest.kt]: What type of error is occurring please be more specific,

```show_location_error_when_service_error_occured```?

File: app/build.gradle [hunk 3]
```diff
@@ -57,4 +62,6 @@ dependencies {
     // Mockito
     testImplementation "org.mockito:mockito-core:2.15.0"
     implementation 'com.google.android.gms:play-services-location:16.0.0'
+
+    testImplementation "com.nhaarman:mockito-kotlin:1.5.0"
```
Comment (N2Android) [inline app/build.gradle]: This is a useful lib - just be careful last updates was made 7 months ago. Should work but just confirm is the best way to use mockito for kotlin.

File: app/src/test/java/com/example/flightstimetable/ui/nearby/NearbyPresenterTest.kt (hunk 2 above)
Comment (rogernkosi) [reply app/src/test/java/com/example/flightstimetable/ui/nearby/NearbyPresenterTest.kt]: is it a location error when service error occurs or it is just simply a service error caused by the current device location?

**Comment by rogernkosi** [comment]
Please note: for the error to occur we must use coordinates which cannot return data, or use an incorrect API key

File: app/src/test/java/com/example/flightstimetable/ui/nearby/NearbyPresenterTest.kt (hunk 2 above)
Comment (N2Android) [reply app/src/test/java/com/example/flightstimetable/ui/nearby/NearbyPresenterTest.kt]: Either way - the test naming can be a little more specific. 

But what makes even less sense is your test setup - success does not mean an error... if you are showing an error on service success your logic is not great.

File: app/build.gradle [hunk 4]
```diff
@@ -57,4 +62,6 @@ dependencies {
     // Mockito
```
Comment (N2Android) [inline app/build.gradle]: Remove this comment as well please.

**Pull request merged**